lib/certificate_chain_serializer.rb

```diff
@@ -0,0 +1,17 @@
+class CertificateChainSerializer
+  DELIMITER = "\n-----END CERTIFICATE-----\n".freeze
+
+  def self.load(chain)
+    return [] unless chain
+    certs = chain.split(DELIMITER)
+    certs.map do |cert|
+      cert += DELIMITER
+      OpenSSL::X509::Certificate.new(cert)
+    end
+  end
+
+  def self.dump(chain)
+    return if chain.blank?
+    chain.map(&:to_s).map(&:strip).join("\n")
```

Comment:
why do we need strip and join here?

Reply:
I was aiming to normalize the newlines from the supplied cert(s). I now replaced the implementation with one that produces normalized output by using `OpenSSL::X509::Certificate` to generate the PEM formatted string…